File: app/src/main/java/com/englishalphabet/SoundPlayer.java
package com.englishalphabet;

import android.content.Context;
import android.media.MediaPlayer;

public class SoundPlayer {

    private MediaPlayer mediaPlayer;

    public void playSound(Context context, int soundResource) {
        stopSound();

        mediaPlayer = MediaPlayer.create(context, soundResource);
        mediaPlayer.start();
    }

    public void stopSound() {
        if (mediaPlayer != null) {
            mediaPlayer.release();
            mediaPlayer = null;
        }
    }
}

File: app/src/main/java/com/englishalphabet/SoundPlayer.kt
package com.englishalphabet

import android.content.Context
import android.media.MediaPlayer

class SoundPlayer {

    private var mediaPlayer: MediaPlayer? = null

    fun playSound(context: Context, soundResource: Int) {
        stopSound()

        mediaPlayer = MediaPlayer.create(context, soundResource)?.apply {
            setOnCompletionListener {
                stopSound()
            }
            start()
        }
    }

    fun stopSound() {
        mediaPlayer?.let { player ->
            try {
                if (player.isPlaying) {
                    player.stop()
                }
            } catch (e: IllegalStateException) {
                e.printStackTrace()
            } finally {
                player.release()
                mediaPlayer = null
            }
        }
    }
}
